src/dfm/data/load_datasets.py

```diff
@@ -0,0 +1,217 @@
+        repeats: int=1,
+        **kwargs
+    ) -> IterableDatasetDict:
+    """
+    Loads DaNews.
+
+    Args:
+        path_to_danews (Union[str, Path], optional): path to the DaNews dataset.
+            Defaults to /work/hope-infomedia_cleaned
+        columns_to_keep (Optional[List[str]], optional): Columns to keep. Default to None
+            in which case all columns are kept.
+        kwargs: arguments to be passed forward to load_dataset
+    
+    Returns:
+        IterableDatasetDict: A datasets IterableDatasetDict
+    """
+    path_to_danews = Path(path_to_danews)
+
+    test = path_to_danews / "infomedia_2000-2021_v1.0.0_test.jsonl"
+    val = path_to_danews / "infomedia_2000-2021_v1.0.0_val.jsonl"
+    train = path_to_danews / "infomedia_2000-2021_filtered_v1.0.0_train.jsonl"
+    data_files = {"train": [str(train)]*repeats, "test": [str(test)], "validation": [str(val)]}
+    danews = load_dataset("json", data_files=data_files, streaming=True)
+    
+    _add_source = partial(__add_source, source = "danews")
+    danews = danews.map(_add_source)
+
+    if columns_to_keep:
+        danews = keep_columns(danews, columns_to_keep)
+    return danews
+
+
+def load_nat(
+        path_to_nat: Union[str, Path]=NAT_PATH, 
+        years: Iterable[int]=range(2006, 2017), 
+        probabilities: Optional[List[float]] = None, 
+        seed: Optional[int] = None,
+        columns_to_keep: Optional[List[str]]=None,
+        repeats: int=10,
+        **kwargs
+    ) -> IterableDatasetDict:
+    """
+    Loads NetArkivet Text corpus (NAT).
+
+    Args:
+        path_to_nat (Union[str, Path], optional): path to the NAT dataset.
+            Defaults to /work/netarkivet-cleaned.
+        columns_to_keep (Optional[List[str]], optional): Columns to keep. Default to None
+            in which case all columns are kept.
+        years (Iterable[int]): A list of years to include.
+        probabilities (Optional[List[float]], optionak): Interleave probabilites of years.
+            Default to None, denoting equal probabilites.
+        kwargs: arguments to be passed forward to load_dataset
+    
+    Returns:
+        IterableDatasetDict: A datasets IterableDatasetDict
+    """
+    path_to_nat = Path(path_to_nat)
+
+    train_paths = []
+    for year in years:
+        train_path = path_to_nat / f"{year}_deduplicated_filtered.jsonl"
+        train_paths.append(str(train_path))
+    datasets = [load_dataset("json", data_files={"train": [path]*repeats}, streaming=True, split="train", **kwargs) 
+                for path in train_paths]
+    nat = interleave_datasets(datasets=datasets, probabilities=probabilities, seed=seed)
+    nat = IterableDatasetDict({"train": nat})
+    
+    _add_source = partial(__add_source, source = "nat")
+    nat = nat.map(_add_source)
+
+    if columns_to_keep:
+        nat = keep_columns(nat, columns_to_keep)
+    return nat
+
+def load_dcc_v1(
+    probabilities=[0.06, 0.06, 0.03, 0.85],
```

Comment:
Hmm, seems to have a very high risk of mixing up the order if changing in the future. 

Perhaps we could pass a list of lists with `[loader_function, path, probability]`, and then unzip that at usage time?

Reply:
Have made a dict